hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -125,10 +125,10 @@ public S3EventsHoodieIncrSource(
     if (!StringUtils.isNullOrEmpty(props.getString(Config.S3_KEY_PREFIX))) {
       filter = filter + " and s3.object.key like '" + props.getString(Config.S3_KEY_PREFIX) + "%'";
     }
-    if (!StringUtils.isNullOrEmpty(props.getString(Config.S3_IGNORE_KEY_PREFIX))) {
+    if (!StringUtils.isNullOrEmpty(props.getString(Config.S3_IGNORE_KEY_PREFIX, null))) {
```

Comment:
can we also fix L 125?

Reply:
We need the folder prefix as a required argument I guess right ? Otherwise incremental job will ingest all the files in the bucket.

Reply:
ok, I was thinking what incase user is interested to pull all data for a given bucket.

Reply:
@codope did you folks have any discussion around this when the 2 stage pipeline support was added ?